wordpress-mobile/gutenberg-mobile
# Feature/1675 dark mode android integration

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/1675

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/19293
WordPress-Android PR: https://github.com/wordpress-mobile/WordPress-Android/pull/11004

![ezgif com-video-to-gif (13)](https://user-images.githubusercontent.com/28219092/71314666-d3ef2100-244c-11ea-811f-7f2aa352feb9.gif)

To test:

On devices with API <=28:
1. Open App Settings from Me screen.
2. Notice the App Theme preference.
3. The default value should be set to `Set by Battery Saver`.
4. From the preference, dialog select `Dark` and confirm that the app switches to Dark Theme.
5. Open the Gutenberg Editor and confirm that it has Dark Theme
6. Select `Light` and confirm that the app switches to Light Theme.
7. Open the Gutenberg Editor and confirm that it has Light Theme
8. Select `Set by Battery Saver` and enable Battery Saver on the device.
9. Confirm that Dark Theme is used when Battery Saver is ON.

On devices with API 29+:
1. Open App Settings from Me screen.
2. Notice the App Theme preference.
3. The default value should be set to `System Default`.
4. From the preference, dialog select `Dark` and confirm that the app switches to Dark Theme.
5. Open the Gutenberg Editor and confirm that it has Dark Theme
6. Select `Light` and confirm that the app switches to Light Theme.
7. Open the Gutenberg Editor and confirm that it has Light Theme
8. Select `System Default` and close the app.
9. Enable Dark Theme in Display settings of the device.
10. Launch the app again and confirm that the dark theme is applied.

Note for API 29+: At this moment we aren't able to get the app theme to switch automatically when the Dark Theme is toggled. For now, you need to restart the app for new theme to take effect. @khaykov has created a separate issue for this at https://github.com/wordpress-mobile/WordPress-Android/issues/10748

PR submission checklist:

- [ ] replace "future release" in release notes with some version
